upgrade/scripts/upgrade_python_package.py

```diff
@@ -216,7 +220,12 @@ def is_cloudsmith_url_valid(cloudsmith_url):
 
 def is_package_already_installed(package):
     results = pip("list", "--format", "json")
-    parsed_results = json.loads(results)
+    try:
```

Comment:
After all of these discussions and worries about format of pip output, errors, pollution of the script execution output, I'm thinking if we can maybe avoid using pip altogether. Maybe `importlib`? https://blog.finxter.com/python-check-version-of-package-with-pip
Though we really should not spend too much more time and effort into addressing some rather unlikely scenarios imo.
Might even be better to revert this implementation to `git show`, despite that shortcoming in order to avoid dropping support for older pip. But again, not sure if we even care about pip version which are so old

Reply:
That looks cool and could potentially be a simpler solution. But as I've looked through `importlib`, it seems that they are new in >=3.8. Docs found here https://docs.python.org/3.8/library/importlib.metadata.html
In 3.6 it raises an error.
```
ModuleNotFoundError: No module named 'importlib.metadata'
```
So I think we'll have to keep going with this approach for a little while into the future, until we completely stop supporting 3.6

Reply:
Right, amid Python 3.10 reworks, I forgot that we still need this script to support Python 3.6. Can you also revert the drop 3.6 support changes in this PR?